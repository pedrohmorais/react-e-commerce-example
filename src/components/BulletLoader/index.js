import React from 'react';
import { oneOf } from 'prop-types';
import StyledBall from './style';

const propTypes = {
  kind: oneOf(['default-loading', 'primary-loading', 'secondary-loading']),
  size: oneOf(['small', 'medium', 'large']),
};

const defaultProps = {
  kind: 'primary-loading',
  size: 'medium',
};

const BulletLoader = ({ size, kind }) => (
  <StyledBall size={size} kind={kind}>
    <div />
    <div />
    <div />
  </StyledBall>
);

BulletLoader.defaultProps = defaultProps;
BulletLoader.propTypes = propTypes;

export default BulletLoader;
