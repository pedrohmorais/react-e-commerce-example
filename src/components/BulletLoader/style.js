import styled from 'styled-components';

import theme from 'styled-theming';
import { colors } from 'theme';

const backgroundColor = theme.variants('mode', 'kind', {
  'default-loading': { main: colors.lipstick },
  'primary-loading': { main: colors.white },
  'secondary-loading': { main: colors.white },
});

const ballSize = theme.variants('mode', 'size', {
  small: { main: '8px' },
  medium: { main: '10px' },
  large: { main: '15px' },
});

const containerSize = theme.variants('mode', 'size', {
  small: { main: '66px' },
  medium: { main: '120px' },
  large: { main: '250px' },
});

export default styled.div`
  @keyframes fadeIn {
    from {
      opacity: 0;
    }
    to {
      opacity: 1;
    }
  }

  @keyframes bounce {
    33% {
      transform: translateY(5px);
    }
    66% {
      transform: translateY(-4px);
    }
  }

  min-width: ${containerSize};
  width: 100%;

  > div {
    background-color: ${backgroundColor};
    width: ${ballSize};
    height: ${ballSize};
    margin: 0 2px;
    border-radius: 100%;
    animation-fill-mode: both;
    display: inline-block;
  }

  > div:nth-child(1) {
    animation: bounce 0.6s -0.14s infinite ease-in-out;
  }

  > div:nth-child(2) {
    animation: bounce 0.6s -0.07s infinite ease-in-out;
  }

  > div:nth-child(3) {
    animation: bounce 0.6s 0s infinite ease-in-out;
  }
`;
