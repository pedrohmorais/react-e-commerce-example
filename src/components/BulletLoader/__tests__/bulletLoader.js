import React from 'react';
import { shallow } from 'enzyme';

import BulletLoader from 'components/BulletLoader';

describe('<BulletLoader />', () => {
  it('should match snapshot', () => {
    const component = <BulletLoader size="small" kind="default-loading" />;

    const wrapper = shallow(component);

    expect(wrapper).toMatchSnapshot();
  });
});
