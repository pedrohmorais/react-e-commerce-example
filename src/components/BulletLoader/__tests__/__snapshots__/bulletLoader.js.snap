// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BulletLoader /> should match snapshot 1`] = `
<styled.div
  kind="default-loading"
  size="small"
>
  <div />
  <div />
  <div />
</styled.div>
`;
